src/views/Login.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { FunctionComponent, useState, useEffect } from 'react';
+import { RouteComponentProps, navigate } from '@reach/router';
+import styled from 'styled-components';
+import { Formik, Field, Form } from 'formik';
+import axios from 'axios';
+
+import { Alert, Button, Heading, Input } from '../components';
+import useSessionStorage from '../hooks/useSessionStorage';
+import { requiredField } from '../utils/validations';
+import { usersEndpoint } from '../utils/endpoints';
+
+type LoginProps = {} & RouteComponentProps;
+
+const LoginWrapper = styled.div`
+  max-width: ${(props) => props.theme.breakpoints.md};
+  margin: 0 auto;
+`;
+
+const StyledLoginFormWrapper = styled(Form)`
+  border: ${(props) => props.theme.mixins.baseBorderStyle};
+  padding: ${(props) => props.theme.sizes.doubleSpacer};
+`;
+
+const Login: FunctionComponent<LoginProps> = () => {
+  const { session, saveSessionStorage } = useSessionStorage('tartanHub');
+  const [isLoading, setIsLoading] = useState(false);
+  const [showErrorMessage, setShowErrorMessage] = useState(false);
+
+  useEffect(() => {
+    if (session) {
+      navigate('/gallery');
+    }
+  }, [session]);
+
+  return (
+    <LoginWrapper>
+      <Heading>Log In</Heading>
+      <Formik
+        validateOnMount
+        initialValues={{ username: '', password: '' }}
+        onSubmit={(values, { setSubmitting, resetForm }) => {
+          setIsLoading(true);
+          setShowErrorMessage(false);
+
+          axios
+            .get(usersEndpoint)
```

Comment:
```suggestion
            .get<Array<{ email: string; username: string }>>(usersEndpoint)
```
Another generics opportunity.